src/test/java/com/techsophy/tsf/workflow/engine/camunda/service/impl/KeycloakClientCredentialsTest.java

```diff
@@ -0,0 +1,80 @@
+package com.techsophy.tsf.workflow.engine.camunda.service.impl;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.techsophy.tsf.workflow.engine.camunda.dto.ClientDetails;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.keycloak.admin.client.Keycloak;
+import org.keycloak.admin.client.resource.ClientResource;
+import org.keycloak.admin.client.resource.ClientsResource;
+import org.keycloak.admin.client.resource.RealmResource;
+import org.keycloak.representations.idm.ClientRepresentation;
+import org.keycloak.representations.idm.CredentialRepresentation;
+import org.mockito.ArgumentMatchers;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.test.util.ReflectionTestUtils;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+@ExtendWith(MockitoExtension.class)
+ class KeycloakClientCredentialsTest {
+
+    @Mock
+    private Keycloak keycloak;
+    @Mock
+    RealmResource realmResource;
+    @Mock
+    ClientsResource clientsResource;
+    @Mock
+    ClientResource clientResource;
+    @Mock
+    private Map<String, ClientDetails> tenantSecretCacheMock;
+    @Mock
+    CredentialRepresentation credentialRepresentation;
+    @InjectMocks
+    private KeycloakClientCredentials keycloakClientCredentials;
+    @BeforeEach
+    public void setUp() {
+        ReflectionTestUtils.setField(keycloakClientCredentials,"clientId","camunda-identity-service");
+    }
+
+    @Test
+    void fetchClientDetailsBooleanFalseArg()
+    {
+        ClientDetails clientDetails = new ClientDetails();
+        clientDetails.setClientId("1212");
+        clientDetails.setSecret("121");
+        Mockito.when(tenantSecretCacheMock.get(ArgumentMatchers.anyString())).thenReturn(clientDetails);
```

Comment:
lets not mock the cache here, as this is part of business logic we are implementing, Discuss with me if its not clear.

Reply:
use a spy here.